CMakeLists.txt

```diff
@@ -142,22 +145,67 @@ add_executable(${PROJECT_NAME} ${GUI_TYPE}
                ${HEADERS}
                ${FORMS}
                ${RESOURCES}
-               ${CMAKE_SOURCE_DIR}/res/windows/resources.rc)
-
-target_include_directories(${PROJECT_NAME} PUBLIC ${CMAKE_CURRENT_SOURCE_DIR} LibREDasm QHexEdit)
-add_dependencies(${PROJECT_NAME} LibREDasm)
-
-if(WIN32)
-    target_link_libraries(${PROJECT_NAME} Qt5::Core Qt5::Gui Qt5::Widgets LibREDasm)
-else()
-    target_link_libraries(${PROJECT_NAME} Qt5::Core Qt5::Gui Qt5::Widgets pthread LibREDasm)
-endif()
+               ${CMAKE_SOURCE_DIR}/res/windows/resources.rc
+)
+
+# TODO: refactor this to target_include_directories(target PUBLIC ...) in the submodules
+target_include_directories(${PROJECT_NAME} PRIVATE
+        ${CMAKE_CURRENT_SOURCE_DIR}
+        LibREDasm
+        QHexEdit
+)
+
+target_link_libraries(${PROJECT_NAME} PRIVATE
+    Qt5::Core
+    Qt5::Gui
+    Qt5::Widgets
+    LibREDasm
+)
+
+# Include Threads
+# TODO: perhaps this should be a PUBLIC dependency of LibREDasm?
+find_package(Threads REQUIRED)
+target_link_libraries(${PROJECT_NAME} PRIVATE ${CMAKE_THREAD_LIBS_INIT})
 
 if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
     target_compile_options(${PROJECT_NAME} PRIVATE -Wall -Wno-missing-braces)
 endif()
 
 # Post build commands
-set(LIBREDASM_BIN ${CMAKE_BINARY_DIR}/LibREDasm/LibREDasm${CMAKE_SHARED_LIBRARY_SUFFIX})
-add_custom_command(TARGET ${PROJECT_NAME} POST_BUILD COMMAND ${CMAKE_COMMAND} -E copy ${LIBREDASM_BIN} ${CMAKE_BINARY_DIR})
-add_custom_command(TARGET ${PROJECT_NAME} POST_BUILD COMMAND ${CMAKE_COMMAND} -E create_symlink submodules plugins)
+add_custom_command(TARGET ${PROJECT_NAME} POST_BUILD COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:LibREDasm> ${CMAKE_BINARY_DIR})
+if(NOT WIN32)
```

Comment:
On Windows you need admin rights to create a symbolic link for a directory. I think a better way to handle this is by using `RUNTIME_OUTPUT_DIRECTORY`: https://stackoverflow.com/a/6595001/1806760

If you do it that way you will also be able to support multi-configuration generators (such as Visual Studio).